tests/callbacks/test_quantization.py

```diff
@@ -138,3 +153,16 @@ def test_quantization_triggers(tmpdir, trigger_fn: Union[None, int, Callable], e
     trainer.fit(qmodel, datamodule=dm)
 
     assert qcb._forward_calls == expected_count
+
+
+def test_non_quantizable(tmpdir):
+    """QAT params can work around quantization limits of complex layers like embeddings"""
+
+    # Training fit works
+    model = NonQuantizableModel()
+    qcb = QuantizationAwareTraining(method_to_quantize='qforward', modules_to_skip=['embed'])
+    trainer = Trainer(callbacks=[qcb], default_root_dir=tmpdir, max_epochs=1)
+    trainer.fit(model)
+
+    # Quant/dequant stubs are in the appropriate place within the converted model
+    model(model.example_input_array)
```

Comment:
Can you add check to make sure embed wasn't quantised ?

Reply:
I was thinking to add an `assert model.embed.qconfig is None`, but all the `qconfig` attributes get stripped by the `torch.quantization.convert()` call in `on_fit_end`. It might still be possible to check the class type of the embedding